crates/rpc-types/src/eth/transaction/receipt.rs

```diff
@@ -69,4 +70,90 @@ impl TransactionReceipt {
         }
         Some(self.from.create(nonce))
     }
+
+    /// Returns `true` if the transaction was successful.
+    /// A transaction is considered successful if the status code is `1`.
+    fn success(&self) -> bool {
+        match &self.status_code {
+            Some(status) => status == &U64::from(1),
+            None => false,
+        }
+    }
+
+    /// Returns the logs emitted by the transaction.
+    /// Converts the logs from the RPC type to the internal type.
+    fn logs(&self) -> Vec<PrimitivesLog<LogData>> {
+        let mut logs = Vec::new();
+        for log in &self.logs {
+            let rpc_log: Log = log.clone();
+            let log_data = LogData::try_from(rpc_log).unwrap_or_default();
```

Comment:
I'm okay with using new_unchecked checked here, because we assume the topics in a receipt are correct

Reply:
fixed https://github.com/alloy-rs/alloy/pull/395/commits/720162dd3b00b5a8c55dcc74573b8f7a2d59649e